src/server/typingsInstaller/nodeTypingsInstaller.ts

```diff
@@ -139,8 +139,8 @@ namespace ts.server.typingsInstaller {
                 if (this.log.isEnabled()) {
                     this.log.writeLine(`npm install #${requestId} took: ${Date.now() - start} ms${sys.newLine}stdout: ${stdout}${sys.newLine}stderr: ${stderr}`);
                 }
-                // treat any output on stdout as success
-                onRequestCompleted(!!stdout);
+                // treat absence of error as success
+                onRequestCompleted(!err);

```

Comment:
I assume we've confirmed that "npm install" returns a non-0 error code across platforms/versions if a package fails to install.